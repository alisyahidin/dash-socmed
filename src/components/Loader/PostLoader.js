import React from 'react'
import ContentLoader from "react-content-loader"

import { random } from './helper'
import './style.scss'

const PostLoader = ({repeat, long, ...props}) => [...Array(repeat)].map((_, i) => (
  <ContentLoader
    key={i}
    className="post-loader"
    height={long ? 190 : 100}
    width={400}
    speed={2}
    primaryColor="#f3f3f3"
    secondaryColor="#ecebeb"
    {...props}
  >
    <rect x="0" y="15" rx="5" ry="5" width="230" height="7" />
    <rect x="0" y="30" rx="3" ry="3" width="85" height="6.4" />
    <rect x="0" y="50" rx="3" ry="3" width={random(200, 350)} height="6.4" />
    <rect x="0" y="65" rx="3" ry="3" width={random(200, 350)} height="6.4" />
    <rect x="0" y="80" rx="3" ry="3" width={random(200, 350)} height="6.4" />
    {long && (
      <>
        <rect x="0" y="95" rx="3" ry="3" width={random(200, 350)} height="6.4" />
        <rect x="0" y="110" rx="3" ry="3" width={random(200, 350)} height="6.4" />
        <rect x="0" y="125" rx="3" ry="3" width={random(200, 350)} height="6.4" />
        <rect x="0" y="140" rx="3" ry="3" width={random(200, 350)} height="6.4" />
        <rect x="0" y="155" rx="3" ry="3" width={random(200, 350)} height="6.4" />
        <rect x="0" y="170" rx="3" ry="3" width={random(200, 350)} height="6.4" />
      </>
    )}
  </ContentLoader>
))

PostLoader.defaultProps = {
    repeat: 1,
    long: false
}

export default PostLoader