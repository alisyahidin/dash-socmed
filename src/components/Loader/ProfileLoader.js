import React from 'react'
import ContentLoader from "react-content-loader"

const ProfileLoader = props => (
  <ContentLoader
    height={400}
    width={400}
    speed={2}
    primaryColor="#f3f3f3"
    secondaryColor="#ecebeb"
    {...props}
  >
    <circle cx="50%" cy="55" r="35" />
    <rect x="30%" y="105" rx="7" ry="7" width="40%" height="13" />
    <rect x="40%" y="130" rx="7" ry="7" width="20%" height="9" />
    <rect x="20" y="175" rx="7" ry="7" width="18%" height="10" />
    <rect x="150" y="175" rx="7" ry="7" width="40%" height="10" />
    <rect x="20" y="200" rx="7" ry="7" width="20%" height="10" />
    <rect x="150" y="200" rx="7" ry="7" width="45%" height="10" />
    <rect x="20" y="225" rx="7" ry="7" width="22%" height="10" />
    <rect x="150" y="225" rx="7" ry="7" width="35%" height="10" />
    <rect x="20" y="250" rx="7" ry="7" width="18%" height="10" />
    <rect x="150" y="250" rx="7" ry="7" width="40%" height="10" />
    <rect x="20" y="275" rx="7" ry="7" width="20%" height="10" />
    <rect x="150" y="275" rx="7" ry="7" width="45%" height="10" />
    <rect x="20" y="300" rx="7" ry="7" width="21%" height="10" />
    <rect x="150" y="300" rx="7" ry="7" width="35%" height="10" />
    <rect x="20" y="325" rx="7" ry="7" width="20%" height="10" />
    <rect x="150" y="325" rx="7" ry="7" width="40%" height="10" />
    <rect x="20" y="350" rx="7" ry="7" width="23%" height="10" />
    <rect x="150" y="350" rx="7" ry="7" width="45%" height="10" />
  </ContentLoader>
)

ProfileLoader.defaultProps = {
    repeat: 1
}

export default ProfileLoader