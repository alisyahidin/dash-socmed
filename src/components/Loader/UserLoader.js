import React from 'react'
import ContentLoader from "react-content-loader"
import { random } from './helper'

const UserLoader = ({repeat, ...props}) => [...Array(repeat)].map((_, i) => (
  <ContentLoader
    key={i}
    height={30}
    width={400}
    speed={2}
    primaryColor="#f3f3f3"
    secondaryColor="#ecebeb"
    {...props}
  >
    <circle cx="10" cy="10" r="10" />
    <rect
      x="25"
      y="7"
      rx="5"
      ry="5"
      height="7"
      width={random(80, 150)}
    />
  </ContentLoader>
))

UserLoader.defaultProps = {
    repeat: 1
}

export default UserLoader