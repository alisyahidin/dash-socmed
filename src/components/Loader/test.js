import React from 'react'
import { shallow } from 'enzyme'
import PostLoader from './PostLoader'
import ProfileLoader from './ProfileLoader'
import UserLoader from './UserLoader'

it('render <PostLoader /> correctly', () => {
  const component = shallow(<PostLoader />)

  expect(component).toMatchSnapshot()
})

it('render <ProfileLoader /> correctly', () => {
  const component = shallow(<ProfileLoader />)

  expect(component).toMatchSnapshot()
})

it('render <UserLoader /> correctly', () => {
  const component = shallow(<UserLoader />)

  expect(component).toMatchSnapshot()
})