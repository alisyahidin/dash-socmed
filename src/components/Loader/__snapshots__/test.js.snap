// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <PostLoader /> correctly 1`] = `
<ContentLoader
  className="post-loader"
  height={100}
  key="0"
  primaryColor="#f3f3f3"
  secondaryColor="#ecebeb"
  speed={2}
  width={400}
>
  <rect
    height="7"
    rx="5"
    ry="5"
    width="230"
    x="0"
    y="15"
  />
  <rect
    height="6.4"
    rx="3"
    ry="3"
    width="85"
    x="0"
    y="30"
  />
  <rect
    height="6.4"
    rx="3"
    ry="3"
    width={286}
    x="0"
    y="50"
  />
  <rect
    height="6.4"
    rx="3"
    ry="3"
    width={202}
    x="0"
    y="65"
  />
  <rect
    height="6.4"
    rx="3"
    ry="3"
    width={269}
    x="0"
    y="80"
  />
</ContentLoader>
`;

exports[`render <ProfileLoader /> correctly 1`] = `
<ContentLoader
  height={400}
  primaryColor="#f3f3f3"
  repeat={1}
  secondaryColor="#ecebeb"
  speed={2}
  width={400}
>
  <circle
    cx="50%"
    cy="55"
    r="35"
  />
  <rect
    height="13"
    rx="7"
    ry="7"
    width="40%"
    x="30%"
    y="105"
  />
  <rect
    height="9"
    rx="7"
    ry="7"
    width="20%"
    x="40%"
    y="130"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="18%"
    x="20"
    y="175"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="40%"
    x="150"
    y="175"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="20%"
    x="20"
    y="200"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="45%"
    x="150"
    y="200"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="22%"
    x="20"
    y="225"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="35%"
    x="150"
    y="225"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="18%"
    x="20"
    y="250"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="40%"
    x="150"
    y="250"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="20%"
    x="20"
    y="275"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="45%"
    x="150"
    y="275"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="21%"
    x="20"
    y="300"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="35%"
    x="150"
    y="300"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="20%"
    x="20"
    y="325"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="40%"
    x="150"
    y="325"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="23%"
    x="20"
    y="350"
  />
  <rect
    height="10"
    rx="7"
    ry="7"
    width="45%"
    x="150"
    y="350"
  />
</ContentLoader>
`;

exports[`render <UserLoader /> correctly 1`] = `
<ContentLoader
  height={30}
  key="0"
  primaryColor="#f3f3f3"
  secondaryColor="#ecebeb"
  speed={2}
  width={400}
>
  <circle
    cx="10"
    cy="10"
    r="10"
  />
  <rect
    height="7"
    rx="5"
    ry="5"
    width={117}
    x="25"
    y="7"
  />
</ContentLoader>
`;
